### PR TITLE
Add nxos_vlans resource module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

modules/network/nxos/nxos_vlans.py

### CODE REVIEW

File: lib/ansible/module_utils/network/nxos/argspec/vlans/vlans.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class VlansArgs(object):
+    """The arg spec for the nxos_vlans module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {'config': {'elements': 'dict',
```
Comment (NilashishC) [inline lib/ansible/module_utils/network/nxos/argspec/vlans/vlans.py:39]: Suggestion: `yapf --in-place </path/to/file>` makes these argspec files really easy to read. :)
https://github.com/google/yapf

**Review by NilashishC — COMMENTED** [review]
It's probably a good idea to use `block` and `always` in the tests, to make sure we're cleaning up even if an assertion fails. This way, the next targets would never be affected.

**Review by NilashishC — CHANGES_REQUESTED** [review]
We should also be asserting the commands sent to the device in the tests.

File: lib/ansible/module_utils/network/nxos/config/vlans/vlans.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                obj_in_have = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+                commands.extend(self.del_all(obj_in_have))
+        else:
+            if not have:
+                return commands
+            for h in have:
+                commands.append('no vlan ' + str(h['vlan_id']))
+        return commands
+
+    def del_all(self, obj):
```
Comment (NilashishC) [inline lib/ansible/module_utils/network/nxos/config/vlans/vlans.py]: Can this method be clubbed with `add_commands` by adding a flag to determine if we need to append a negate?
Like [this](https://github.com/ansible/ansible/pull/59724/files#diff-9b7e2c51c0a9dcacfabf900240548efaR239).

**Comment by trishnaguha** [comment]
> We should also be asserting the commands sent to the device in the tests.

@NilashishC Added.

File: test/integration/targets/nxos_vlans/tests/cli/overridden.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  - name: Overridden
+    nxos_vlans: &overridden
+      config:
+        - vlan_id: 9
+          name: test-vlan9
+          enabled: false
+      state: overridden
+    register: result
+
+  - assert:
```
Comment (NilashishC) [inline test/integration/targets/nxos_vlans/tests/cli/overridden.yaml:37]: We also need to assert that no extra command is sent.

Comment (trishnaguha) [reply test/integration/targets/nxos_vlans/tests/cli/overridden.yaml:37]: Fixed

File: lib/ansible/modules/network/nxos/nxos_vlans.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  returned: when changed
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
```
Comment (NilashishC) [inline lib/ansible/modules/network/nxos/nxos_vlans.py]: A couple of commands here would be good.

File: lib/ansible/module_utils/network/nxos/argspec/facts/facts.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 CHOICES = [
     'all',
     'lag_interfaces',
+    'vlans',
```
Comment (NilashishC) [inline lib/ansible/module_utils/network/nxos/argspec/facts/facts.py:14]: We also need to add `!{{ network_resource }}`?

**Review by NilashishC — APPROVED** [review]
+1

**Pull request merged**